core/src/lib/output_formatter_resource.h

```diff
@@ -0,0 +1,148 @@
+                      std::string baseformat = "}\n");
+
+  void KeyBool(const char* name, bool value, bool as_comment = false);
+
+  void KeySignedInt(const char* name, int64_t value, bool as_comment = false);
+  void KeyUnsignedInt(const char* name, int64_t value, bool as_comment = false);
+
+  void KeyString(const char* name, const char* value, bool as_comment = false)
+  {
+    KeyUnquotedString(name, value, as_comment);
+  }
+  void KeyString(const char* name, std::string value, bool as_comment = false)
+  {
+    KeyUnquotedString(name, value.c_str(), as_comment);
+  }
+  void KeyQuotedString(const char* name,
+                       const char* value,
+                       bool as_comment = false);
+  void KeyQuotedString(const char* name,
+                       std::string value,
+                       bool as_comment = false);
+  void KeyUnquotedString(const char* name,
+                         const char* value,
+                         bool as_comment = false);
+  void KeyUnquotedString(const char* name,
+                         std::string value,
+                         bool as_comment = false);
+
+  void KeyMultipleStringsInOneLine(const char* key,
+                                   alist* list,
+                                   bool as_comment = false,
+                                   bool quoted_strings = true);
+
+  void KeyMultipleStringsInOneLine(const char* key,
+                                   alist* list,
+                                   const char* (*GetValue)(void* item),
+                                   bool as_comment = false,
+                                   bool quoted_strings = true);
+
+  void KeyMultipleStringsOnePerLine(const char* key,
+                                    alist& list,
+                                    bool as_comment = false,
+                                    bool quoted_strings = true);
+
+  void KeyMultipleStringsOnePerLine(const char* key,
+                                    alist* list,
+                                    bool as_comment = false,
+                                    bool quoted_strings = true);
+
+  void KeyMultipleStringsOnePerLine(const char* key,
+                                    alist* list,
+                                    const char* (*GetValue)(void* item),
+                                    bool as_comment = false,
+                                    bool quoted_strings = true);
+
+  void KeyMultipleStringsOnePerLine(const char* key,
+                                    const std::vector<std::string>&,
+                                    bool as_comment = false,
+                                    bool quoted_strings = true);
+
+  void ArrayStart(const char* key,
+                  bool as_comment = false,
+                  std::string baseformat = "%s {\n");
+  void ArrayEnd(const char* key,
+                bool as_comment = false,
+                std::string baseformat = "}\n");
```

Comment:
Same applies to baseformat of ArrayEnd. Could Eliminate this function parameter completely.

Reply:
It is set to `""` in some cases.